router/api.go

```diff
@@ -55,22 +55,31 @@ func (api *API) CreateRoute(ctx context.Context, w http.ResponseWriter, req *htt
 
 	err := l.AddRoute(route)
 	if err != nil {
-		if err == ErrConflict {
-			rjson, err := json.Marshal(&route)
-			if err != nil {
-				log.Error(err.Error())
-				httphelper.Error(w, err)
-				return
-			}
-			httphelper.Error(w, httphelper.JSONError{
-				Code:    httphelper.ConflictErrorCode,
-				Message: "Duplicate route",
-				Detail:  rjson,
-			})
+		rjson, jerr := json.Marshal(&route)
+		if jerr != nil {
+			log.Error(jerr.Error())
+			httphelper.Error(w, jerr)
 			return
 		}
-		log.Error(err.Error())
-		httphelper.Error(w, err)
+		var message string
+		var code httphelper.ErrorCode
```

Comment:
Instead of these variables, we could do one `JSONError` variable and assign into the struct fields directly.